homeassistant/components/directv/media_player.py

```diff
@@ -112,7 +112,7 @@ async def async_update(self) -> None:
             self._paused = self._last_position == self._program.position
             self._is_recorded = self._program.recorded
             self._last_position = self._program.position
-            self._last_update = state.at
+            self._last_update = dt_util.utcnow()
```

Comment:
This does not look correct? Seems it should be fixed in lib so that the state.at is corrected.

Reply:
> This does not look correct? Seems it should be fixed in lib so that the state.at is corrected.

State.at is never updated in the lib.  I don't think the intent of State.at in the lib is to represent the last time the position or any other attribute was updated and this component is incorrectly using state.at to update the media_position_updated_at attribute.  The State dataclass in the lib has the frozen=True decorator.  Other media player components are also using utcnow() to update the media_position_updated_at attribute.

Reply:
Yup, indeed the lib looks wrong. The at value would be a static value, not related to state object construction. Maybe @ctalkington  want to fix that, but I think this change is okey for now.